core/bindings.rs

```diff
@@ -186,24 +183,22 @@ pub fn set_func(
 
 // Register a raw v8::FunctionCallback
 // with some external data.
-#[allow(clippy::too_many_arguments)]
 pub fn set_func_raw(
   scope: &mut v8::HandleScope<'_>,
   obj: v8::Local<v8::Object>,
   name: &'static str,
   callback: v8::FunctionCallback,
   external_data: *const c_void,
   fast_function: &Option<Box<dyn FastFunction>>,
-  snapshot_loaded: bool,
-  will_snapshot: bool,
+  snapshot_options: SnapshotOptions,
 ) {
   let key = v8::String::new(scope, name).unwrap();
   let external = v8::External::new(scope, external_data as *mut c_void);
   let builder =
     v8::FunctionTemplate::builder_raw(callback).data(external.into());
   let templ = if let Some(fast_function) = fast_function {
     // Don't initialize fast ops when snapshotting, the external references count mismatch.
-    if snapshot_loaded && !will_snapshot {
+    if snapshot_options == SnapshotOptions::Load {
```

Comment:
I think this was wrong originally - probably should check if snapshot was loaded or there was no snapshot at all

Reply:
ok